libkv.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"log"
+	"strings"
+
+	"github.com/blang/vfs"
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+	"github.com/docker/libkv/store/boltdb"
+	"github.com/docker/libkv/store/consul"
+	"github.com/docker/libkv/store/etcd"
+	"github.com/docker/libkv/store/zookeeper"
+)
+
+// logFatal is defined so log.Fatal calls can be overridden for testing
+var logFatal = log.Fatal
+
+// LibKV -
+type LibKV struct {
+	store store.Store
+	fs    vfs.Filesystem
+}
+
+// NewLibKV - instantiate a new
+func NewLibKV(url string) *LibKV {
+	env := &Env{}
+
+	var source_type store.Backend
+	var client string
+
+	source_type = ""
+	client = ""
+
+	if strings.HasPrefix(url, "consul:") {
```

Comment:
If you change the function's signature to accept a `*url.URL`, you can switch on `url.Scheme` instead - would probably be a little less verbose, and cleaner